lib/Driver/ToolChains.cpp

```diff
@@ -347,7 +349,10 @@ ToolChain::constructInvocation(const CompileJobAction &job,
       Arguments.push_back(context.getAllSourcesPath());
     } else {
       for (const Action *A : context.InputActions) {
-        cast<InputAction>(A)->getInputArg().render(context.Args, Arguments);
+        const Arg &InputArg = cast<InputAction>(A)->getInputArg();
+        // Frontend no longer tolerates redundant input files on command line.
+        if (InputArg.getValue() != PrimaryFileWrittenToCommandLine)
+          InputArg.render(context.Args, Arguments);
```

Comment:
Aha. This looks like a mistake from the SourceKit folks: changing the order of files might change the way the file is processed. (It *shouldn't,* of course, but that's not where we are today, and there will always be small discrepancies because of imported Clang modules.)

This does seem like a consistent change, but it's probably not the right thing longterm. @benlangmuir, who owns -index-file-path now that Ashley's gone?

Reply:
Should I assume that I can merge the PR with this code in it?

Reply:
Yep, this one's okay.